File: hw/suarez/U3/HW18AbstractFactory/src/ec/edu/espe/model/GUIFactory.java
package ec.edu.espe.model;

/**
 *
 * @author bernardo
 */

public abstract class GUIFactory {
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
        if (sys == 1)
            return new WinFactory();
        else
            return new LinuxFactory();
    }

    private static int readFromConfigFile(String key) {
        return 0; // mock: 0 = Windows, 1 = Linux
    }

    public abstract Button createButton();
    public abstract Menu createMenu();
}

File: hw/paillacho/U3/HW19AbstractFactory/AbstractFactoryPython/ec/edu/espe/composite/abstractfactory/controller/GUIFactory.py
"""
Abstract GUI Factory

@author PC
"""
from abc import ABC, abstractmethod
import platform


class GUIFactory(ABC):
    
    @staticmethod
    def get_factory():
        sys_type = GUIFactory._read_from_config_file("OS_TYPE")
        if sys_type == 0:
            from .WinFactory import WinFactory
            return WinFactory()
        else:
            from .LinuxFactory import LinuxFactory
            return LinuxFactory()
    
    @staticmethod
    def _read_from_config_file(key):
        # Simplified config reading - returns 0 for Windows, 1 for Linux
        os_name = platform.system().lower()
        if "win" in os_name:
            return 0
        else:
            return 1
    
    @abstractmethod
    def create_button(self):
        pass
    
    @abstractmethod
    def create_menu(self):
        pass
